toolchain/lower/file_context.h

```diff
@@ -219,6 +301,37 @@ class FileContext {
 
   // Global format string for `printf.int.format` used by the PrintInt builtin.
   llvm::Value* printf_int_format_string_ = nullptr;
+
+  // For a generic function, keep track of the specifics for which LLVM
+  // function declarations were created. Those can be retrieved then from
+  // specific_functions_. We resize this to the correct size. Vector indexes
+  // correspond to `GenericId` indexes.
+  llvm::SmallVector<llvm::SmallVector<SemIR::SpecificId>, 0> lowered_specifics_;
+
+  // For specifics that exist in lowered_specifics, a hash of their function
+  // type information: return and parameter types. We resize this to the
+  // correct size. Vector indexes correspond to `SpecificId` indexes.
+  // TODO: Hashing all members of FunctionTypeInfo may not be necessary.
+  llvm::SmallVector<llvm::BLAKE3Result<32>, 0>
+      lowered_specifics_type_fingerprint_;
+
+  // This is initialized and populated while lowering a specific.
+  // We resize this to the correct size. Vector indexes correspond to
+  // `SpecificId` indexes.
+  llvm::SmallVector<SpecificFunctionFingerprint, 0>
+      lowered_specific_fingerprint_;
+
+  // Equivalent specifics that have been found. For each specific, this points
+  // to the canonical equivalent specific, which may also be self. We resize
+  // this to the correct size and initialize to SpecificId::None, which defines
+  // that there is no other equivalent specific to this SpecificId. Vector
+  // indexes correspond to `SpecificId` indexes.
+  llvm::SmallVector<SemIR::SpecificId> equivalent_specifics_;
```

Comment:
```suggestion
  llvm::SmallVector<SemIR::SpecificId, 0> equivalent_specifics_;
```

Similar to the above?